Profiler/CProfilerCallback.cpp

```diff
@@ -359,41 +347,44 @@ int CProfilerCallback::writeFileVersionInfo(LPCWSTR assemblyPath, char* buffer,
 
 HRESULT CProfilerCallback::JITCompilationFinished(FunctionID functionId,
 	HRESULT hrStatus, BOOL fIsSafeToBlock) {
-	FunctionInfo info;
-	getFunctionInfo(functionId, &info);
+	recordFunctionInfo(&jittedMethods, LOG_KEY_JITTED, functionId);
+	return S_OK;
+}
 
-	if (isEagerMode) {
-		writeSingleFunctionInfoToLog(LOG_KEY_JITTED, info);
-	}
-	else {
-		// Notify monitor that method has been jitted.
-		jittedMethods.push_back(info);
+HRESULT CProfilerCallback::JITInlining(FunctionID callerID, FunctionID calleeId,
+		BOOL* pfShouldInline) {
+	// Save information about inlined method (if not already seen)
+	if (inlinedMethodIds.insert(calleeId).second == true) {
+		recordFunctionInfo(&inlinedMethods, LOG_KEY_INLINED, calleeId);
 	}
 
-	// Always return OK
+	// Always allow inlining.
+	*pfShouldInline = true;
+
 	return S_OK;
 }
 
-HRESULT CProfilerCallback::JITInlining(FunctionID callerID, FunctionID calleeId,
-		BOOL* pfShouldInline) {
+void CProfilerCallback::recordFunctionInfo(std::vector<FunctionInfo>* recordedFunctionInfos, const char* logKey, FunctionID calleeId) {
 	FunctionInfo info;
 	getFunctionInfo(calleeId, &info);
 
-	if (isEagerMode) {
-		writeSingleFunctionInfoToLog(LOG_KEY_INLINED, info);
-	}
-	else {
-		// Save information about inlined method.
-		if (inlinedMethodIds.insert(calleeId).second == true) {
-			inlinedMethods.push_back(info);
+	EnterCriticalSection(&criticalSection);
+
+	if (eagerness == 1) {
+		// Directly write to log if we want to record each function immediatelly
+		// This is solely for performance reasons as we are writing data live
+		writeSingleFunctionInfoToLog(logKey, info);
+	} else {
+		// otherwise record function info
+		recordedFunctionInfos->push_back(info);

```

Comment:
these inline comments (also below) seem redundant. please remove. for the one below: how about extracting a method for the condition check with a speaking name?

Reply:
good idea